app/src/org/commcare/views/widgets/ComboboxWidget.java

```diff
@@ -115,9 +116,17 @@ public void afterTextChanged(Editable s) {
 
     private void fillInPreviousAnswer(FormEntryPrompt prompt) {
         if (prompt.getAnswerValue() != null) {
-            String previousAnswerValue = ((Selection)prompt.getAnswerValue().getValue()).getValue();
-            if (comboBox.isValidUserEntry(previousAnswerValue)) {
-                comboBox.setText(previousAnswerValue);
+            Selection priorSelection = (Selection)prompt.getAnswerValue().getValue();
+            String previousEnteredText;
+            try {
+                previousEnteredText = prompt.getSelectItemText(priorSelection);
+            } catch (XPathTypeMismatchException e) {
+                // Means that a default value was provided that is not a precise answer choice,
+                // but we can still grab the entered text this way and check if it's a valid entry
+                previousEnteredText = priorSelection.xmlValue;
```

Comment:
Will this ever be expected to be a valid selection? xmlValue here is always the <value> choice, right? I don't think we want to be searching for the <value> body if it's not in the current list of valid choices.

You could imagine a scenerio where the label for a value matched the semantic value of another element, even, in which case this would overwrite that answer.

I think it'd be good to coordinate this logic to be consistent with the other select options if possible, seems odd to have different codepaths for each one.